nixos/modules/tasks/notify-updates.nix

```diff
@@ -0,0 +1,60 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.system.notifyUpdates;
+
+in {
+  options = {
+    system.notifyUpdates = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to notify the user when there are NixOS updates available
+        '';
+      };
+
+      channel = mkOption {
+        type = types.str;
+        example = "https://nixos.org/channels/nixos-unstable";
+        description = ''
+          The URI of the NixOS channel to use for checking updates.
+          By default, this is the channel set using
+          <command>nix-channel</command> (run <literal>nix-channel
+          --list</literal> to see the current value).
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    systemd.user.services.nixos-notify-updates = {
+      description = "Notify available NixOS Updates";
+
+      restartIfChanged = true;
+
+      serviceConfig.Type = "oneshot";
+
+      environment = config.nix.envVars // {
+        inherit (config.environment.sessionVariables) NIX_PATH;
+      };
+
+      path = with pkgs; [ coreutils curl libnotify ];
```

Comment:
```suggestion
      path = with pkgs; [ coreutils curl gnused libnotify ];
```

Reply:
Thanks (i still have to properly test it beyond just copying the script into my config)